crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -854,14 +854,24 @@ impl<'src> Parser<'src> {
         })
     }
 
+    /// Parses a unary expression.
+    ///
+    /// This includes the unary arithmetic `+` and `-`, bitwise `~`, and the
+    /// boolean `not` operators.
+    ///
+    /// # Panics
+    ///
+    /// If the parser isn't positioned at any of the unary operators.
+    ///
+    /// See: <https://docs.python.org/3/reference/expressions.html#unary-arithmetic-and-bitwise-operations>
     pub(super) fn parse_unary_expression(&mut self) -> ast::ExprUnaryOp {
         let start = self.node_start();
 
         let op = UnaryOp::try_from(self.current_token_kind())
-            .expect("Expected operator to be a unary operator token.");
+            .expect("current token should be a unary operator");
         self.bump(self.current_token_kind());
```

Comment:
Nit: What do you think, would it make sense to make the expression accept an optional `UnaryOp` argument and have a `unary_op.token_kind()` method? 

It would remove the need for `.expect`

Reply:
I thought of a few ideas where the simplest was to use the pattern:
```rs
let op = if self.eat('+') {
	UnaryOp::UAdd
} ... 
else {
	self.bump('not'); 
	UnaryOp::Not
};
```

What you're suggesting would be to change the call site to pass the correct unary operator instead. That would still involve having multiple branches and repeating the error handling. I'll think about this but for now I'll merge this as is.